x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/phases/warm_phase.tsx

```diff
@@ -27,47 +27,64 @@ import {
   OptionalLabel,
   ErrableFormRow,
   SetPriorityInput,
-  NodeAllocation,
   MinAgeInput,
+  DataTierAllocation,
+  DescribedFormField,
+  NodesDataProvider,
+  DefaultAllocationWarning,
 } from '../components';
 
-const shrinkLabel = i18n.translate('xpack.indexLifecycleMgmt.warmPhase.shrinkIndexLabel', {
-  defaultMessage: 'Shrink index',
-});
-
-const moveToWarmPhaseOnRolloverLabel = i18n.translate(
-  'xpack.indexLifecycleMgmt.warmPhase.moveToWarmPhaseOnRolloverLabel',
-  {
-    defaultMessage: 'Move to warm phase on rollover',
-  }
-);
-
-const forcemergeLabel = i18n.translate('xpack.indexLifecycleMgmt.warmPhase.forceMergeDataLabel', {
-  defaultMessage: 'Force merge data',
-});
+const i18nTexts = {
+  shrinkLabel: i18n.translate('xpack.indexLifecycleMgmt.warmPhase.shrinkIndexLabel', {
+    defaultMessage: 'Shrink index',
+  }),
+  forcemergeLabel: i18n.translate('xpack.indexLifecycleMgmt.warmPhase.forceMergeDataLabel', {
+    defaultMessage: 'Force merge data',
+  }),
+  moveToWarmPhaseOnRolloverLabel: i18n.translate(
+    'xpack.indexLifecycleMgmt.warmPhase.moveToWarmPhaseOnRolloverLabel',
+    {
+      defaultMessage: 'Move to warm phase on rollover',
+    }
+  ),
+  defaultAllocationNotAvailable: {
+    title: i18n.translate(
+      'xpack.indexLifecycleMgmt.warmPhase.dataTier.defaultAllocationNotAvailableTitle',
+      { defaultMessage: 'No warm tier nodes found' }
```

Comment:
I'm wondering if we want to avoid "XYZ tier nodes" as well. Is there any benefit over just saying "No warm nodes found." or "No nodes assigned to the warm tier"

Reply:
Great point! I like "No nodes assigned to the warm tier" as it alludes to something the user can do too.

Reply:
This is where I've landed based on the current round of feedback for the warning:

<img width="1120" alt="Screenshot 2020-09-09 at 16 15 24" src="https://user-images.githubusercontent.com/8155004/92610223-beaa7300-f2b7-11ea-87e3-7b21aa364d7f.png">